solutions/usaco-531.mdx

```diff
@@ -0,0 +1,124 @@
+---
+id: usaco-531
+source: USACO Silver 2015 February
+title: Superbull
+author: Maggie Liu
+---
+
+[Official Analysis](http://www.usaco.org/current/data/sol_superbull_silver.html)
+
+## Explanation
+
+We can use Kruskal's to find the maximum spanning tree of the graph formed by
+the teams. First, add all possible edges between two teams to the vector
+$\texttt{edges}$ and sort it in decreasing order of edge weight. For each edge,
+if we can unite the connected components of the two vertices, then we have
+eliminated one team. Increment $\texttt{eliminated}$ by $1$ and add the edge
+weight to our answer, $\texttt{ans}$. Once $n - 1$ teams have been eliminated,
+we can print $\texttt{ans}$.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N^2 \log N)$
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <cstdio>
+#include <algorithm>
+#include <vector>
+using namespace std;
+void init(int n);
+bool unite(int a, int b);
```

Comment:
why not just idk
put the actual function definitions here

Reply:
idk, I like to have the main function at the top since that's the general idea of the code (like if I had a solve function that was doing the work and a main function that only called the solve function t times, I would put the solve function at the top)

Reply:
then maybe codesnip it?

Reply:
Yeah, that's a good idea